ci: sanitycheck: enable asserts again

nashif opened the pull request:
We were running without sanitycheck options in the final and important
run.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
recheck

ydamigos commented:
Could you also fix the typo in commit's header?

nashif commented:
> Could you also fix the typo in commit's header?

done

pfalcon commented:
recheck

pfalcon commented:
> recheck

Was just checking that codecov.io reports jump back and forth in their number so far. Indeed, they do.

nashif commented:
> Was just checking that codecov.io reports jump back and forth in their number so far. Indeed, they do.

that is usually the case when first job reports and then another job with coverage data is merged in which changes the numbers. This is annoying and might be confusing indeed. Per their documentation, a delay is added for matrix jobs, but that is not the case it seems.

pfalcon commented:
> that is usually the case when first job reports and then another job with coverage data is merged in which changes the numbers.

I figure we just don't have stable enough overall coverage config, and that contributes to it too.

nashif committed:
ci: run sanitycheck with assertions enabled

This option was dropped by mistake, reenabling now..

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: filter on CONFIG_ASSERT

CONFIG_ASSERT is being set by cmake, so it is not possible to filter
using the generated config, add this as a standalone filter.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
kernel: remove empty string in assert statement

This was failing with compiler warnings. Looks like latest compilers
enable warnings by default that we do not have in the current SDK.

This was failing with unit tests being built natively.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: rpl: do not build when assert is enabled

Was failing due to increased size of binary.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: enable coverage

coverage option was dropped by mistake moving away from IS_TEST.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
